compute/cloud-client/src/main/java/compute/custommachinetype/CreateWithHelper.java

```diff
@@ -0,0 +1,307 @@
+ * limitations under the License.
+ */
+
+package compute.custommachinetype;
+
+// [START compute_custom_machine_type_create_with_helper]
+
+import com.google.cloud.compute.v1.AttachedDisk;
+import com.google.cloud.compute.v1.AttachedDiskInitializeParams;
+import com.google.cloud.compute.v1.InsertInstanceRequest;
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstancesClient;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Operation;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.IntStream;
+
+public class CreateWithHelper {
+
+  // Returns the array of integers within the given range, incremented by the specified step.
+  // start (inclusive): starting number of the range
+  // stop (inclusive): ending number of the range
+  // step : increment value
+  static int[] getNumsInRangeWithStep(int start, int stop, int step) {
+    return IntStream.range(start, stop).filter(x -> (x - start) % step == 0).toArray();
+  }
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "your-google-cloud-project-id";
+    // Name of the zone to create the instance in. For example: "us-west3-b".
+    String zone = "google-cloud-zone";
+    // Name of the new virtual machine (VM) instance.
+    String instanceName = "instance-name";
+    String cpuSeries = "N1";
+    // Number of CPU cores you want to use.
+    int coreCount = 2;
+    // The amount of memory for the VM instance, in megabytes.
+    int memory = 256;
+
+    createInstanceWithCustomMachineTypeWithHelper(projectId, zone, instanceName, cpuSeries,
+        coreCount, memory);
+  }
+
+  public static String customMachineTypeUri(String zone, String cpuSeries, int coreCount,
+      int memory) {
+
+    if (!Arrays.asList(CpuSeries.E2.cpuSeries, CpuSeries.N1.cpuSeries, CpuSeries.N2.cpuSeries,
+        CpuSeries.N2D.cpuSeries).contains(cpuSeries)) {
+      throw new Error(String.format("Incorrect cpu type: %s", cpuSeries));
+    }
+
+    TypeLimits typeLimit = Objects.requireNonNull(
+        typeLimitsMap.get(CpuSeries.get(cpuSeries).name())).typeLimits;
+
+    // Perform the following checks to verify if the requested parameters are allowed.
+    // Find more information about limitations of custom machine types at:
+    // https://cloud.google.com/compute/docs/general-purpose-machines#custom_machine_types
+    //
```

Comment:
(nit) remove the commenting on this empty line. But leave this space between the above comment and the one below